common/scala/src/main/scala/org/apache/openwhisk/core/database/ActivationStore.scala

```diff
@@ -154,6 +162,23 @@ trait ActivationStore {
     since: Option[Instant] = None,
     upto: Option[Instant] = None,
     context: UserContext)(implicit transid: TransactionId): Future[Either[List[JsObject], List[WhiskActivation]]]
+
+  /**
+   * Check if the system is configured to not store the activation in the database
+   * Dont't store if activation is successful, blocking, not in debug mode and no disable store is configured
```

Comment:
Trying to understand this, I found it easier to apply De Morgan's law :) hence I suggest
```suggestion
   * Only stores activations if one of these is true:
   * - result is an error,
   * - a non-blocking activation
   * - an activation in debug mode
   * - activation stores is not disabled via a configuration parameter
```